### PR TITLE
Add public Vale configuration

### PR DESCRIPTION
**Pull request by jdbaldry**
Open sources our Vale configuration to make it more easily used by external contributors.

Also updates CI to test repository documentation against repository Vale configuration.

In a future PR:
- [ ] Add image packaging to CI
- [ ] Add VSCode extension documentation.

### CODE REVIEW

File: docs/sources/write/style-guide/ux-writing/index.md [hunk 1]
```diff
@@ -158,12 +158,12 @@ Using long blocks of descriptive text reduces readability. Write important infor
 
 > Before you begin, make sure you have the following items:
 >
-> - The kubectl command-line tool. To learn how to install kubectl, refer to [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl).
+> - The `kubectl` command-line tool. To learn how to install `kubectl`, refer to [`kubectl`](https://kubernetes.io/docs/tasks/tools/#kubectl).
 > - (Optional) The helm command-line tool for managing Helm charts. To learn how to install helm, refer to [Installing Helm](https://helm.sh/docs/intro/install/) in the Helm documentation.
```
Comment (josmperez) [inline docs/sources/write/style-guide/ux-writing/index.md:162]:
```suggestion
> - (Optional) The `helm` command-line tool for managing Helm charts. To learn how to install helm, refer to [Installing Helm](https://helm.sh/docs/intro/install/) in the Helm documentation.
```

Comment (josmperez) [reply docs/sources/write/style-guide/ux-writing/index.md:162]: Not sure if it's helm or `helm`, but helm unformatted looks wrong.

File: docs/sources/write/style-guide/style-conventions/index.md [hunk 2]
```diff
@@ -67,11 +67,11 @@ Make content relevant to the user's context. The more familiar you are with the
 
 When you write in present tense, you avoid words such as have, has, had, been, should, would, and will.
```
Comment (josmperez) [inline docs/sources/write/style-guide/style-conventions/index.md:68]:
```suggestion
When you write in present tense, avoid words such as have, has, had, been, should, would, and will.
```

File: docs/sources/write/shortcodes/index.md [hunk 3]
```diff
@@ -176,7 +177,10 @@ The content within the shortcode tags is as follows:
...
 - `project path prefix` - Designates the target project. In the example above, the path prefixes are `/docs/grafana/` for Grafana and `/docs/grafana-cloud/` for Cloud.
-- `reference` - The path to the destination file. It can include `<SOMETHING VERSION>`, which is either taken from front matter of (which file) or falls back to being inferred from the version of the page. This enables the use of absolute paths that resolve correctly, irrespective of version. When including a version, for the target project, use the name of the project, with spaces but no hyphens or underscores, all upper-cased (for example, grafana = GRAFANA, grafana-cloud = GRAFANA CLOUD).
+- `reference` - The path to the destination file. It can include `<SOMETHING VERSION>`, which is either taken from front matter of the page or falls back to being inferred from the version of the page.
+  This enables the use of absolute paths that resolve correctly, irrespective of version.
+  When including a version, for the target project, use the name of the project, with spaces but no hyphens or underscores, all upper-cased.
```
Comment (josmperez) [inline docs/sources/write/shortcodes/index.md:182]:
```suggestion
  When including a version, for the target project, use the name of the project, with spaces but no hyphens or underscores, all upper-case.
```

File: docs/sources/write/shortcodes/index.md [hunk 4]
```diff
@@ -21,7 +21,7 @@ Markdown is limited in its ability to render complex elements. Although you migh
 
 The following sections describe shortcodes available for use in Grafana Markdown files. To learn about other shortcodes, refer to the Hugo [shortcode documentation](https://gohugo.io/content-management/shortcodes/).
 
-> **Note for internal Grafana Labs contributors**: The Grafana shortcode templates are defined in the `layouts/shortcodes` folder of the website repo. To request custom shortcodes, [create an issue](https://github.com/grafana/writers-toolkit/issues).
+> **Note for internal Grafana Labs contributors**: The Grafana shortcode templates are defined in the `layouts/shortcodes` folder of the website repository. To request custom shortcodes, [create an issue](https://github.com/grafana/writers-toolkit/issues).
```
Comment (josmperez) [inline docs/sources/write/shortcodes/index.md:24]: ironically, does not use admonition shortcode.

File: docs/sources/write/references/index.md [hunk 5]
```diff
@@ -174,4 +172,4 @@ In this example,
 - `:14` represents the line number containing the unresolved reference
 - `:47` represents the character in that line where the unresolved reference begins
 
-If the reference's destination appears to be invalid, for example due to a typo in the reference or relref directory traversal depth, then you should be able to resolve this by correcting the reference target.
+If the reference's destination appears to be invalid, for example due to a typo in the reference or `relref` directory traversal depth, then you should be able to resolve this by correcting the reference target.
```
Comment (josmperez) [inline docs/sources/write/references/index.md:175]:
```suggestion
If the reference's destination appears to be invalid, for example due to a typo in the reference or the depth of the`relref` directory, then you should be able to resolve this by correcting the reference target.
```

File: docs/sources/write/image-guidelines/index.md [hunk 6]
```diff
@@ -193,14 +194,18 @@ General rules:
...
 - Windows: Use [Snagit](https://www.techsmith.com/screen-capture.html)
 - Web browser: Use [Photopea](https://www.photopea.com/)
 
+<!-- vale Grafana.Spelling = YES -->
+
 ## Where to store media assets
 
-All visual assets are stored in Google Cloud Storage, only accessible to Grafana Labs employees. You use the [asset uploader application](https://admin.grafana.com/upload/) to upload assets.
+All visual assets are stored in Google Cloud Storage, only accessible to Grafana Labs employees. You use the [asset upload application](https://admin.grafana.com/upload/) to upload assets.
 The following table lists the steps you take to provide the Grafana Labs technical documentation team with the image.
```
Comment (josmperez) [inline docs/sources/write/image-guidelines/index.md:209]:
```suggestion
The following table lists the steps you take to provide us with the image.
```

File: docs/sources/write/front-matter/index.md [hunk 7]
```diff
@@ -246,6 +236,13 @@ To redirect the page `/docs/grafana/latest/alerting/unified-alerting/` to `/docs
 
 Therefore, with the alias `./alerting/unified-alerting/` in the source file for the page `/docs/grafana/latest/alerting/`, Hugo will create a redirect page at `/docs/grafana/latest/alerting/unified-alerting/`.
 
+#### Other projects
+
+Include an `aliases` entry for the current URL path.
+Adding an `aliases` entry makes it safer to move content around as the redirect from old to new page location is already in place.
```
Comment (josmperez) [inline docs/sources/write/front-matter/index.md:242]:
```suggestion
Addi an `aliases` entry to make it safer to move content around, as the redirect from old to new page location is already in place.
```

File: docs/sources/write/front-matter/index.md [hunk 8]
```diff
@@ -123,19 +123,19 @@ cascade:
...
+### `menuTitle`
 
 Use to specify a different heading in the sidebar navigation than the `title` element, for example if you want to abbreviate the topic heading in the table of contents.
 
-### title (required)
+### `title` (required)
 
 Hugo uses the `title` to generate the sidebar table of contents if there is no `menuTitle` specified in the front matter. If the `doc-validator` linter has been implemented on your repository, your topic heading must exactly match the title in the metadata.
 
 The `title` becomes the document title element in the HTML. Often browsers display this in the tab for the page.
```
Comment (josmperez) [inline docs/sources/write/front-matter/index.md:134]:
```suggestion
The `title` becomes the document title element in the HTML. Often, browsers display this in the tab for the page.
```

File: docs/sources/write/front-matter/index.md [hunk 9]
```diff
@@ -123,19 +123,19 @@ cascade:
       - cloud
 ```
 
-### menuTitle
+### `menuTitle`
 
 Use to specify a different heading in the sidebar navigation than the `title` element, for example if you want to abbreviate the topic heading in the table of contents.
```
Comment (josmperez) [inline docs/sources/write/front-matter/index.md:128]:
```suggestion
Use to specify a different heading in the sidebar navigation than the `title` element; for example, if you want to abbreviate the topic heading in the table of contents.
```

File: docs/sources/write/front-matter/index.md [hunk 10]
```diff
@@ -72,19 +72,19 @@ Often, this doesn’t need to be original text, you can often scan the first few
 If it's too long, it is harmlessly truncated on social media.
 Use double quotes (`"`) to surround the title. Do not use smart quotes.
 
-### draft
+### `draft`
 
 When set to `true`, this option prevents Hugo from rendering the content.
 Use the command line flag `--buildDrafts` to generate content marked as `draft: true`.
```
Comment (josmperez) [inline docs/sources/write/front-matter/index.md:78]:
```suggestion
Use the command-line flag `--buildDrafts` to generate content marked as `draft: true`.
```

File: docs/sources/write/front-matter/index.md [hunk 11]
```diff
@@ -62,7 +62,7 @@ For example, `date: "2023-04-24T00:00:00Z"` is 12:00 AM, Apr 24 Coordinated Univ
 The `date` front matter also impacts menu ordering.
 Pages with more recent dates are lower in the menu.
 
-### description (required)
+### `description` (required)
 
 Use to provide the short description of the topic to search engines, including the search engine used in the Grafana documentation site. The description is also displayed on social media, such as Twitter, to provide a clue to users about the page contents.
```
Comment (josmperez) [inline docs/sources/write/front-matter/index.md:67]: change name to X?

File: docs/sources/structure/topic-types/tutorial/index.md [hunk 12]
```diff
@@ -130,18 +130,19 @@ To hide your tutorial from your documentation's table of contents:
    ```
 
    {{%admonition type="note" %}}
-   Create the `cascade` field if it does not exist. Substitute your repo for `grafana-cloud` in this example.
+   Create the `cascade` field if it does not exist.
```
Comment (josmperez) [inline docs/sources/structure/topic-types/tutorial/index.md:133]:
```suggestion
   Create the `cascade` field if it doesn't exist.
```

**Review by josmperez — APPROVED** [review]
I've made some suggestions, but nothing critical. LGTM. Good pothole fixing!

File: docs/sources/write/front-matter/index.md (hunk 11 above)
Comment (jdbaldry) [reply docs/sources/write/front-matter/index.md:67]: Lol! I'm not sure Twitter has lost meaning yet. At some point this change might be needed but I don't think it's necessary yet.

File: docs/sources/write/image-guidelines/index.md (hunk 6 above)
Comment (jdbaldry) [reply docs/sources/write/image-guidelines/index.md:209]: I'm not sure if a reader will have enough context to understand who "us" is.

That being said, I'm not sure how the reader knows who "the Grafana Labs technical documentation team" is either.

File: docs/sources/write/shortcodes/index.md (hunk 4 above)
Comment (jdbaldry) [inline docs/sources/write/shortcodes/index.md:24]:
```suggestion
{{% admonition type="note" %}}
The Grafana shortcode templates are defined in the `layouts/shortcodes` folder of the website repository which is only accessible to Grafana Labs employees. 
To request custom shortcodes, [create an issue](https://github.com/grafana/writers-toolkit/issues).
{{% /admonition %}}
```

Comment (jdbaldry) [reply docs/sources/write/shortcodes/index.md]: Resolved with https://github.com/grafana/writers-toolkit/pull/287/files/a68ca46bd455483904d867009c11c3c1fd7a6ca4#r1281553659

File: vale/.vale.ini [hunk 13]
```diff
@@ -0,0 +1,8 @@
+StylesPath = /etc/vale/styles
+MinAlertLevel = suggestion
+
+Packages = https://github.com/jdbaldry/Hugo/releases/download/v0.2.0-rc.1/Hugo.zip
```
Comment (heitortsergent) [inline vale/.vale.ini:4]: I'm curious if we could add the Google package here, and separate what are the Google rules and the Grafana rules? (I'm also not sure if we have a lot of different rules from the Google package?)

But also as a nit, don't mean to stop this from getting the PR merged. :shipit:

Comment (jdbaldry) [reply vale/.vale.ini]: Done in [da2adc5](https://github.com/grafana/writers-toolkit/pull/287/commits/da2adc547f2d66ed9b65c73f17d4bef5d77b0abd)

Comment (heitortsergent) [reply vale/.vale.ini]: Nice! :shipit:

**Pull request merged**